lib/gc/ExecutionEngine/CPURuntime/Microkernel/BrgemmOnednn.cpp

```diff
@@ -57,6 +57,53 @@ static std::vector<brgemm_desc_t> g_brgemm_desc_list;
 static std::vector<brgemm_kernel_t *> g_brgemm_kernel_list;
 static std::vector<std::unique_ptr<char[]>> g_brgemm_palette;
 
+struct brgemm_cache_info_t {
+  std::shared_ptr<brgemm_desc_t> desc;
+  std::shared_ptr<brgemm_kernel_t> kernel;
+  std::shared_ptr<char> palette;
```

Comment:
we could use `shared_ptr<char[]>` for `palette`, and for `desc` and `kernel` we don't need to change, since they are not managed by smart ptr, and storing ptr to vector elements is dangerous as well

Reply:
fixed.